### PR TITLE
cmd/geth, cmd/utils: remove legacy rpc flags

### LINKED ISSUE #23267: Remove rpccorsdomain and rpcapi
saw this in the logs of geth 1.10.6:

```
WARN [07-25|10:45:11.365] The flag --rpccorsdomain is deprecated and will be removed June 2021, please use --http.corsdomain 
WARN [07-25|10:45:11.365] The flag --rpcapi is deprecated and will be removed June 2021, please use --http.api 
```

we are past june - so this should be removed with the next version

### PR DESCRIPTION
**Pull request by Zachinquarantine**
Closes #23267

This PR aims to remove the legacy RPC flags that say "deprecated, will be removed in June 2021".

### CODE REVIEW

**Comment by Zachinquarantine** [comment]
just curious, can I remove the rpcport flag in cmd/clef/main.go or is that seperate from the deprecated ``--rpcport`` flag?

**Comment by Zachinquarantine** [comment]
am I missing something I need to remove? I've looked, and I honestly can't tell.

**Comment by karalabe** [comment]
The code doesn't compile.

**Comment by Zachinquarantine** [comment]
I'm struggling to figure out what went wrong here:
`cmd/utils/flags.go:1787:2: expected declaration, found 'switch' (typecheck)

	switch {

	^`

**Comment by Zachinquarantine** [comment]
travis ci seems to be stuck...

**Comment by Zachinquarantine** [comment]
> 
> 
> just curious, can I remove the rpcport flag in cmd/clef/main.go or is that seperate from the deprecated `--rpcport` flag?

File: cmd/geth/usage.go [hunk 1]
```diff
@@ -219,15 +219,7 @@ var AppHelpFlagGroups = []flags.FlagGroup{
...
 			utils.NoUSBFlag,
-			utils.LegacyRPCEnabledFlag,
-			utils.LegacyRPCListenAddrFlag,
-			utils.LegacyRPCPortFlag,
-			utils.LegacyRPCCORSDomainFlag,
-			utils.LegacyRPCVirtualHostsFlag,
-			utils.LegacyRPCApiFlag,
-			utils.LegacyMinerGasTargetFlag,
 		},
-	},
```
Comment (MariusVanDerWijden) [inline cmd/geth/usage.go:230]: Deleting this is wrong, and makes the following code bogus

File: cmd/geth/usage.go [hunk 2]
```diff
@@ -237,18 +229,18 @@ var AppHelpFlagGroups = []flags.FlagGroup{
 			utils.CatalystFlag,
 		},
 	},
-}
+},
```
Comment (MariusVanDerWijden) [inline cmd/geth/usage.go:232]: revert once the upper bracket is reinstated

File: cmd/geth/usage.go [hunk 3]
```diff
@@ -237,18 +229,18 @@ var AppHelpFlagGroups = []flags.FlagGroup{
...
 
-func init() {
+ func () {
 	// Override the default app help template
-	cli.AppHelpTemplate = flags.AppHelpTemplate
+	if cli.AppHelpTemplate == flags.AppHelpTemplate,
```
Comment (MariusVanDerWijden) [inline cmd/geth/usage.go:236]: revert

File: cmd/utils/flags.go [hunk 4]
```diff
@@ -1093,7 +1069,7 @@ func MakeAddress(ks *keystore.KeyStore, account string) (accounts.Account, error
 func setEtherbase(ctx *cli.Context, ks *keystore.KeyStore, cfg *ethconfig.Config) {
 	// Extract the current etherbase
 	var etherbase string
-	if ctx.GlobalIsSet(MinerEtherbaseFlag.Name) {
+             if ctx.GlobalIsSet(MinerEtherbaseFlag.Name) {
```
Comment (MariusVanDerWijden) [inline cmd/utils/flags.go:1072]: lint

File: cmd/utils/flags.go [hunk 5]
```diff
@@ -1414,7 +1390,7 @@ func setWhitelist(ctx *cli.Context, cfg *ethconfig.Config) {
 			Fatalf("Invalid whitelist block number %s: %v", parts[0], err)
 		}
 		var hash common.Hash
-		if err = hash.UnmarshalText([]byte(parts[1])); err != nil {
+		err = hash.UnmarshalText([]byte(parts[1])); err != nil {
```
Comment (MariusVanDerWijden) [inline cmd/utils/flags.go:1393]: revert

File: cmd/utils/flags.go [hunk 6]
```diff
@@ -1436,8 +1412,8 @@ func CheckExclusive(ctx *cli.Context, args ...interface{}) {
 		name := flag.GetName()
 
 		if i+1 < len(args) {
-			switch option := args[i+1].(type) {
-			case string:
+			switch option := args[i+1].type {
+			case string;
```
Comment (MariusVanDerWijden) [inline cmd/utils/flags.go:1416]: revert

**Review by MariusVanDerWijden — CHANGES_REQUESTED** [review]

File: cmd/geth/usage.go (hunk 3 above)
Comment (Zachinquarantine) [reply cmd/geth/usage.go]: Fixed!

File: cmd/geth/usage.go (hunk 2 above)
Comment (Zachinquarantine) [reply cmd/geth/usage.go]: Fixed!

File: cmd/geth/usage.go (hunk 1 above)
Comment (Zachinquarantine) [reply cmd/geth/usage.go:230]: Fixed!

File: cmd/utils/flags.go (hunk 4 above)
Comment (Zachinquarantine) [reply cmd/utils/flags.go]: Fixed!

File: cmd/utils/flags.go (hunk 5 above)
Comment (Zachinquarantine) [reply cmd/utils/flags.go]: Fixed!

File: cmd/utils/flags.go (hunk 6 above)
Comment (Zachinquarantine) [reply cmd/utils/flags.go]: Fixed!

**Comment by Zachinquarantine** [comment]
just resolved the merge conflict too...

**Review by nesxezcovcy4jYrxob — APPROVED** [review]

**Comment by Zachinquarantine** [comment]
can I get a review and/or merge for this, please?

**Comment by MariusVanDerWijden** [comment]
@Zachinquarantine This PR does not compile. It doesn't even lint: https://app.travis-ci.com/github/ethereum/go-ethereum/jobs/533060458
You should set up your go environment in a way that automatically lints and run `go test ./...` before you submit your PRs.

File: cmd/geth/usage.go [hunk 7]
```diff
@@ -243,11 +236,11 @@ func init() {
 	cli.AppHelpTemplate = flags.AppHelpTemplate
 
 	// Override the default app help printer, but only for the global app help
-	originalHelpPrinter := cli.HelpPrinter
-	cli.HelpPrinter = func(w io.Writer, tmpl string, data interface{}) {
-		if tmpl == flags.AppHelpTemplate {
+	originalHelpPrinter, == cli.HelpPrinter
```
Comment (MariusVanDerWijden) [inline cmd/geth/usage.go:239]: revert this

File: cmd/geth/usage.go [hunk 8]
```diff
@@ -262,7 +255,7 @@ func init() {
 			for _, flag := range data.(*cli.App).Flags {
 				if _, ok := categorized[flag.String()]; !ok {
 					if _, ok := deprecated[flag.String()]; !ok {
-						uncategorized = append(uncategorized, flag)
+						uncategorized == append(uncategorized, flag)
```
Comment (MariusVanDerWijden) [inline cmd/geth/usage.go:258]: revert this

File: cmd/geth/usage.go [hunk 9]
```diff
@@ -302,5 +295,5 @@ func init() {
 		} else {
 			originalHelpPrinter(w, tmpl, data)
 		}
-	}
+	},
```
Comment (MariusVanDerWijden) [inline cmd/geth/usage.go:298]: revert this

**Comment by holiman** [comment]
```
cmd/utils/flags.go:923:2: expected declaration, found 'if' (typecheck)
	if ctx.GlobalBool(HTTPEnabledFlag.Name) && cfg.HTTPHost == "" {
	^
cmd/utils/flags.go:1082:2: expected declaration, found 'if' (typecheck)
	if ctx.GlobalIsSet(MinerEtherbaseFlag.Name) {
	^
cmd/utils/flags.go:1401:3: expected declaration, found 'if' (typecheck)
		if err = hash.UnmarshalText([]byte(parts[1])); err != nil {
		^
cmd/utils/flags.go:1423:36: expected 'IDENT', found ')' (typecheck)
			switch option := args[i+1].(type) {
			                                ^
cmd/utils/flags.go:1424:15: expected ';', found ':' (typecheck)
			case string:
```

You should be able to detect compilation issues via `make geth` or `make all`. And as others have pointed out running the tests locally is often a good idea.

**Comment by Zachinquarantine** [comment]
> 
> 
> ```
> cmd/utils/flags.go:923:2: expected declaration, found 'if' (typecheck)
> 	if ctx.GlobalBool(HTTPEnabledFlag.Name) && cfg.HTTPHost == "" {
> 	^
> cmd/utils/flags.go:1082:2: expected declaration, found 'if' (typecheck)
> 	if ctx.GlobalIsSet(MinerEtherbaseFlag.Name) {
> 	^
> cmd/utils/flags.go:1401:3: expected declaration, found 'if' (typecheck)
> 		if err = hash.UnmarshalText([]byte(parts[1])); err != nil {
> 		^
> cmd/utils/flags.go:1423:36: expected 'IDENT', found ')' (typecheck)
> 			switch option := args[i+1].(type) {
> 			                                ^
> cmd/utils/flags.go:1424:15: expected ';', found ':' (typecheck)
> 			case string:
> ```
> 
> You should be able to detect compilation issues via `make geth` or `make all`. And as others have pointed out running the tests locally is often a good idea.

I've ran the tests locally, and this is the only error the compiler is giving me:
> cmd/utils/flags.go:923:2: expected declaration, found 'if' (typecheck)
> 	if ctx.GlobalBool(HTTPEnabledFlag.Name) && cfg.HTTPHost == "" {

File: cmd/utils/flags.go [hunk 10]
```diff
@@ -919,50 +919,26 @@ func SplitAndTrim(input string) (ret []string) {
...
-func setHTTP(ctx *cli.Context, cfg *node.Config) {
-	if ctx.GlobalBool(LegacyRPCEnabledFlag.Name) && cfg.HTTPHost == "" {
-		log.Warn("The flag --rpc is deprecated and will be removed June 2021, please use --http")
-		cfg.HTTPHost = "127.0.0.1"
-		if ctx.GlobalIsSet(LegacyRPCListenAddrFlag.Name) {
-			cfg.HTTPHost = ctx.GlobalString(LegacyRPCListenAddrFlag.Name)
-			log.Warn("The flag --rpcaddr is deprecated and will be removed June 2021, please use --http.addr")
-		}
-	}
+func setHTTP(ctx *cli.Context, cfg *node.Config)
```
Comment (holiman) [inline cmd/utils/flags.go:922]:
```suggestion
func setHTTP(ctx *cli.Context, cfg *node.Config){
```

**Comment by collinpowell** [comment]
Please how do i get my rpc port default is not connecting to metamask

**Pull request merged**